frontend/src/components/datasets/ShareGroupDatasetModal.tsx

```diff
@@ -0,0 +1,142 @@
+		options.push(group_option);
+    	});
+	console.log("group optioins are", options);
+	console.log("it is of type", typeof(options));
+
+	return (
+		<Container>
+			<Dialog open={open} onClose={handleClose} fullWidth={true} maxWidth="md"
+				sx={{
+					".MuiPaper-root": {
+						padding: "2em",
+					},
+				}}>
+				<DialogTitle>Share dataset &apos;{datasetName}&apos;</DialogTitle>
+				<Divider />
+				<DialogContent>
+					<Typography>Invite groups to collaborate</Typography>
+					<div style={{
+						display: "flex",
+						alignItems: "center"
+					}}>
+						<FormControl variant="outlined" sx={{ m: 1, minWidth: 120 }}>
+							<InputLabel id="demo-simple-select-label">Group</InputLabel>
+							<Select
+                                	labelId="demo-simple-select-label"
+                                	id="demo-simple-select"
+								value={group}
+								// defaultValue={"641214ebfe6405bc949730c6"}
+								label="Group"
+                                	onChange={(event, value) => {
+                                		console.log(event, value);
+                                		console.log("chose a new group");
+                                		console.log(event.target.value);
+                                    		setGroup(event.target.value);
+                                	}}
+                            	>
+								{options.map((group) => (
+            					<MenuItem
+										key={group.label}
+										value={group.value}
+									>
+										{group.label}
+            					</MenuItem>
+								))}
+                            	</Select>
+						</FormControl>
+						<FormControl variant="outlined" sx={{ m: 1, minWidth: 120 }}>
+							<InputLabel id="demo-simple-select-label">Status</InputLabel>
+							<Select
+								labelId="demo-simple-select-label"
+								id="demo-simple-select"
+								value={role}
+								defaultValue={"viewer"}
+								label="Status"
+								onChange={(event, value) => {
+                                	console.log(event, value);
+									setRole(event.target.value);
+								}}
+							>
+								<MenuItem value="owner">Owner</MenuItem>
+								<MenuItem value="editor">Editor</MenuItem>
+								<MenuItem value="uploader">Uploader</MenuItem>
+								<MenuItem value="viewer">Viewer</MenuItem>
+							</Select>
+						</FormControl>
+					</div>
+					<Button variant="contained" sx={{ marginTop: 1 }} onClick={onShare} disabled={(group.length > 0) ? false : true}>Share</Button>
+					<Collapse in={showSuccessAlert}>
+						<br/>
+						<Alert
+							severity="success"
+							action={
+								<IconButton
+									aria-label="close"
+									color="inherit"
+									size="small"
+									onClick={() => {
+										setShowSuccessAlert(false);
+									}}
+								>
+									<CloseIcon fontSize="inherit" />
+								</IconButton>
+							}
+							sx={{ mb: 2 }}
+						>
+                        Successfully added role!
```

Comment:
This is what the success dialog looks like
<img width="1624" alt="Screenshot 2023-03-31 at 7 21 00 AM" src="https://user-images.githubusercontent.com/28983148/229119261-4df1f267-e193-4893-9400-f26b77562a71.png">